### PR TITLE
Make pauseCompaction act as sync call by waiting for compaction thread stopped

### PR DESCRIPTION
**Pull request by wangtao724**
@bellofreedom 

This change is to make ```pauseCompaction``` blocking until compaction thread exit. Please help review.

It is supposed the thread will exit after finish compaction to current file, so didn't add a timeout.

### CODE REVIEW

**Comment by bellofreedom** [comment]
I think current solution is to add some doc/comments to let the user know they need check the if compaction job was really stopped.  In long term, we could design a new mechanism with quick switch if there're active compacted files.

File: src/main/java/com/oath/halodb/HaloDBInternal.java [hunk 1]
```diff
@@ -312,6 +312,14 @@ void setIOErrorFlag() throws IOException {
 
     void pauseCompaction() throws IOException {
         compactionManager.pauseCompactionThread();
```
Comment (amannaly) [inline src/main/java/com/oath/halodb/HaloDBInternal.java]: Got to check HaloDB after a long time and noticed this PR. 

`pauseCompactionThread` already waits for compaction thread to stop before returning. 
https://github.com/yahoo/HaloDB/blob/b81eeacc9dddd7ceb14eef7bbe726550d320dbc5/src/main/java/com/oath/halodb/CompactionManager.java#L66

Therefore is this change really required?

Comment (wangtao724) [reply src/main/java/com/oath/halodb/HaloDBInternal.java]: Got point. Yes. I didn't see that . Will double confirm by checking production log  and close this PR

Comment (wangtao724) [reply src/main/java/com/oath/halodb/HaloDBInternal.java]: @amannaly You are right. Thanks!
```
=2019-08-13 01:32:19,540  INFO [pool-5-thread-25] (com.oath.halodb.CompactionManager) - Pausing compaction thread ...
=2019-08-13 01:32:19,545  INFO [CompactionThread] (com.oath.halodb.CompactionManager) - Compaction thread stopped.
```
